components/combobox/stories/template.js

```diff
@@ -0,0 +1,138 @@
+import { html } from 'lit-html';
+import { classMap } from 'lit-html/directives/class-map.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+
+import { Template as Menu } from '@spectrum-css/menu/stories/template.js';
+import { Template as TextField } from '@spectrum-css/textfield/stories/template.js';
+import { Template as Popover } from '@spectrum-css/popover/stories/template.js';
+import { Template as PickerButton } from '@spectrum-css/pickerbutton/stories/template.js';
+import { Template as ProgressCircle } from '@spectrum-css/progresscircle/stories/template.js';
+
+import { useArgs, useGlobals } from "@storybook/client-api";
+
+import "../index.css";
+
+export const Template = ({
+  rootClass = "spectrum-Combobox",
+  id,
+  content,
+  customClasses = [],
+  size = "m",
+  isOpen = true,
+  isInvalid = false,
+  isValid = false,
+  isQuiet = false,
+  isDisabled = false,
+  isFocused = false,
+  isKeyboardFocused = false,
+  isLoading = false,
+  isRequired = false,
+  readOnly = false,
+  ...globals
+}) => {
+  const [_, updateArgs] = useArgs();
+  const [{lang}] = useGlobals();
+
+  const { express } = globals;
+
+  try {
+    if (!express) import(/* webpackPrefetch: true */ "../themes/spectrum.css");
+    else import(/* webpackPrefetch: true */ "../themes/express.css");
+  } catch (e) {
+    console.warn(e);
+  }
+
+  return html`
+    <div
+      class=${classMap({
+        [rootClass]: true,
+        [`${rootClass}--size${size?.toUpperCase()}`]: typeof size !== "undefined",
+        'is-open': !isDisabled && isOpen,
+        [`${rootClass}--quiet`]: isQuiet,
+        'is-invalid': !isDisabled && isInvalid,
+        'is-valid': !isDisabled && isValid,
+        'is-focused': !isDisabled && isFocused,
+        'is-keyboardFocused': !isDisabled && isKeyboardFocused,
+        'is-loading': isLoading,
+        'is-disabled': isDisabled,
+        ...customClasses.reduce((a, c) => ({ ...a, [c]: true }), {}),
+      })}
+      id=${ifDefined(id)}
+    >
+      ${[
+        TextField({
+          ...globals,
+          size,
+          isQuiet,
+          isDisabled,
+          isValid,
+          isInvalid,
+          isFocused,
+          isKeyboardFocused,
+          customClasses: [
+            `${rootClass}-textfield`,
+            ...(isLoading ? ['is-loading'] : []),
+          ],
+          customInputClasses: [`${rootClass}-input`],
+          contentAfterInput: isLoading ? ProgressCircle({ 
```

Comment:
Oh interesting, so this is being added inside the Textfield wrapper (assuming to help with alignment). I'd like to see this get added to the textfield component specifically rather than as a generic input (meaning we don't want users to put anything inside the textfield container but this specifically is allowed).

Let's do this - let's add isLoading as a property on Textfield and if true, render the progress circle. Probably also need to add a customProgressCircleClasses array to Textfield as well so you can pass down the spectrum-Combobox support.

Reply:
Thanks, I was wondering about this. I originally made it isLoading, and then realized it needed the custom class, and erred on the side of generic. I agree it's probably better to not leave this option wide open w/o another use case of needing something else there. I'll update that.

Reply:
I've updated the story props/controls in [c56fb3f](https://github.com/adobe/spectrum-css/pull/1556/commits/c56fb3f8c438a934ef77d5714a0283cab26eb3dc)